opensrp-reveal/src/main/java/org/smartregister/reveal/repository/RevealRepository.java

```diff
@@ -128,7 +132,9 @@ public synchronized SQLiteDatabase getReadableDatabase(String password) {
 
     @Override
     public synchronized SQLiteDatabase getWritableDatabase(String password) {
-        if (writableDatabase == null || !writableDatabase.isOpen()) {
+        if (StringUtils.isBlank(password)) {
+            throw new IllegalStateException("Password is blank");
+        } else if (writableDatabase == null || !writableDatabase.isOpen()) {
             if (writableDatabase != null) {
                 writableDatabase.close();
```

Comment:
Why do we close the db if we are sure it cannot be open if it is NOT null?

Reply:
Same issue here https://github.com/OpenSRP/opensrp-client-reveal/blob/3a7fea8e6ca701af5796af46cbf6371ec97cf3cc/opensrp-reveal/src/main/java/org/smartregister/reveal/repository/RevealRepository.java#L121.

Reply:
This is not in the scope of the PR. 

However if the db is not open i.e its not null but its not  readable/writable, close is doing cleanup and releasing any resources. This database cannot be opened as opening is a private method. The only way to open it is to call `net.sqlcipher.database.SQLiteOpenHelper#getReadableDatabase(java.lang.String)` or `net.sqlcipher.database.SQLiteOpenHelper#getWritableDatabase(java.lang.String)`

However one must first close the database before trying to open it it a readable/writable mode

Reply:
What is being improved in this pr is logic to do with accessing the db, so it's arguable that it is within scope.

Reply:
```
public void close() {

        if (!isOpen()) {
            return; // already closed
        }
        lock();
        try {
            closeClosable();
            // close this database instance - regardless of its reference count value
            onAllReferencesReleased();
        } finally {
            unlock();
        }
    }
```

This is the code called on `close()`,  if the db `!open` it just returns, so why close in the first place?